### PR TITLE
Fix for i18n minor issues #538

### PR DESCRIPTION
**Pull request by vjrj**
Attached screenshot of my tests of #538 fixes:

image
image
image
image
image
image

Tested with and without using the downloading plugin.

I also:
- added `.param` to some messages using now `{0}` arguments in order to invalidate the crowdin current translations (of ALA and Australia).
- removed a pair of messages that it seems that they are not used now and confuse the translators

### CODE REVIEW

File: grails-app/taglib/au/org/ala/biocache/hubs/OccurrenceTagLib.groovy [hunk 1]
```diff
@@ -299,7 +299,11 @@ class OccurrenceTagLib {
                                     mkp.yieldUnescaped("&nbsp;")
                                 }
                                 span(class: "facet-item") {
-                                    if (fieldResult.i18nCode) {
+                                    // If we have a translation, we use it, if not we try to use the label translation
+                                    // and if not, directly use the label. If the label is missing, use "unknown"
+                                    // In search.js this is done a bit differently:
+                                    // https://github.com/AtlasOfLivingAustralia/biocache-hubs/blob/00f263640edd802d10a071f5d09d146eaa24af34/grails-app/assets/javascripts/search.js#L1946
+                                    if (fieldResult.i18nCode && alatag.message(code: fieldResult.i18nCode) != fieldResult.i18nCode ) {
```
[the remark this answers is not in the document]
Comment (nickdos) [reply grails-app/taglib/au/org/ala/biocache/hubs/OccurrenceTagLib.groovy:306]: @vjrj still waiting on a reply to my comment above...

[the remark this answers is not in the document]
Comment (vjrj) [reply grails-app/taglib/au/org/ala/biocache/hubs/OccurrenceTagLib.groovy:306]: Hi @nickdos thanks for the review. First of all let me show you how we are right now:

I added this dirty debug suffix:
```
-                                        mkp.yield(alatag.message(code: fieldResult.i18nCode, default: fieldResult.label))
+                                        mkp.yield(alatag.message(code: fieldResult.i18nCode, default: fieldResult.label) +  " 1ST-IF")
                                     } else {
-                                        mkp.yield(alatag.message(code: fieldResult.label ?: 'unknown', default: fieldResult.label))
+                                        mkp.yield(alatag.message(code: fieldResult.label ?: 'unknown', default: fieldResult.label) + " 2ND-IF")
```

and in our case, we don't see that the second if it's used:

image

and now with this PR + the same debug info:

image

This is why I added the comparison (similar to what `search.js`) do to some fallback looking for translations with different approaches).

I didn't touch the `unknown` part in this PR, so If you prefer, we can use the same `alatag.message` code that in the first `if`.

[the remark this answers is not in the document]
Comment (nickdos) [reply grails-app/taglib/au/org/ala/biocache/hubs/OccurrenceTagLib.groovy:306]: OK I see I didn't read the `mkp.yield()` code properly - you are right about the `unknown` value. I'll merge the PR - thanks for clarifying that.

**Review by nickdos — APPROVED** [review]

File: grails-app/taglib/au/org/ala/biocache/hubs/OccurrenceTagLib.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (vjrj) [reply grails-app/taglib/au/org/ala/biocache/hubs/OccurrenceTagLib.groovy:306]: Thanks you for the review @nickdos .

**Pull request merged**